src/crypto/vss.rs

```diff
@@ -0,0 +1,270 @@
+
+        let mut pk = [0u8; 33];
+        pk.copy_from_slice(&self.sender_public_key.to_bytes()[..]);
+        size += pk.consensus_encode(&mut s)?;
+
+        let mut pk = [0u8; 33];
+        pk.copy_from_slice(&self.receiver_public_key.to_bytes()[..]);
+        size += pk.consensus_encode(&mut s)?;
+
+        let len = self.positive_commitments.len() as u16;
+        let mut x = [0u8; 2];
+        x.copy_from_slice(&hex::decode(format!("{:0>4x}", len)).map_err(|_| encode::Error::ParseFailed("len"))?);
+        size += x.consensus_encode(&mut s)?;
+
+        for c in &self.positive_commitments {
+            size += c.consensus_encode(&mut s)?;
+        }
+        let mut secret = [0u8; 32];
+        let hex = format!("{:0>64}", self.positive_secret.to_big_int().to_hex());
+        secret.copy_from_slice(&hex::decode(hex).map_err(|_| encode::Error::ParseFailed("positive_secret"))?[..]);
+        size += secret.consensus_encode(&mut s)?;
+
+        for c in &self.negative_commitments {
+            size += c.consensus_encode(&mut s)?;
+        }
+        let mut secret = [0u8; 32];
+        let hex = format!("{:0>64}", self.negative_secret.to_big_int().to_hex());
+        secret.copy_from_slice(&hex::decode(hex).map_err(|_| encode::Error::ParseFailed("positive_secret"))?[..]);
+        size += secret.consensus_encode(&mut s)?;
+        
+        Ok(size)
+    }
+}
+
+impl Decodable for Vss {
+    fn consensus_decode<D: io::Read>(mut d: D) -> Result<Vss, encode::Error> {
+        let bytes: [u8; 33] = Decodable::consensus_decode(&mut d)?;
+        let sender_public_key = PublicKey::from_slice(&bytes[..]).map_err(|_| encode::Error::ParseFailed("sender_public_key"))?;
+
+        let bytes: [u8; 33] = Decodable::consensus_decode(&mut d)?;
+        let receiver_public_key = PublicKey::from_slice(&bytes[..]).map_err(|_| encode::Error::ParseFailed("receiver_public_key"))?;
+
+        let bytes: [u8; 2] = Decodable::consensus_decode(&mut d)?;
+        let length: u16 = ((bytes[0] as u16) << 8) + (bytes[1] as u16);
+        
+        let positive_commitments: Vec<Commitment> = (0..length).flat_map(|_| Decodable::consensus_decode(&mut d)).collect::<Vec<Commitment>>();
+        
+        let bytes: [u8; 32] = Decodable::consensus_decode(&mut d)?;
+        let positive_secret = ECScalar::from(&BigInt::from(&bytes[..]));
+        let negative_commitments: Vec<Commitment> = (0..length).flat_map(|_| Decodable::consensus_decode(&mut d)).collect::<Vec<Commitment>>();
+
+        let bytes: [u8; 32] = Decodable::consensus_decode(&mut d)?;
+        let negative_secret = ECScalar::from(&BigInt::from(&bytes[..]));
+        Ok(Vss::new(sender_public_key, 
+            receiver_public_key, 
+            positive_commitments, 
+            positive_secret, 
+            negative_commitments, 
+            negative_secret,
+        ))
+    }
+}
+
+impl Encodable for Commitment {
+    fn consensus_encode<S: io::Write>(&self, mut s: S) -> Result<usize, encode::Error> {
```

Comment:
How about using 33bytes compressed public key format. It decrease data size 31bytes.

Reply:
For now, Commitment is converted to GE and GE is used internally.
I think it is natural to use 64 bytes format for encoding/decoding GE.

Reply:
I see. keep as it is.

Using 33bytes format has a pro but it is not big problem for now.